src/base/asyncfilestorage.cpp

```diff
@@ -34,53 +34,16 @@
 #include "base/utils/fs.h"
 #include "base/utils/io.h"
 
-QHash<Path, std::weak_ptr<QFile>> AsyncFileStorage::m_reservedPaths;
-QReadWriteLock AsyncFileStorage::m_reservedPathsLock;
-
 AsyncFileStorage::AsyncFileStorage(const Path &storageFolderPath, QObject *parent)
     : QObject(parent)
     , m_storageDir(storageFolderPath)
 {
     Q_ASSERT(m_storageDir.isAbsolute());
 
-    const Path lockFilePath = m_storageDir / Path(u"storage.lock"_s);
-
-    {
-        const QReadLocker readLocker {&m_reservedPathsLock};
-        m_lockFile = m_reservedPaths.value(lockFilePath).lock();
-    }
-
-    if (!m_lockFile)
-    {
-        const QWriteLocker writeLocker {&m_reservedPathsLock};
-        if (std::weak_ptr<QFile> &lockFile = m_reservedPaths[lockFilePath]; lockFile.expired()) [[likely]]
-        {
-            if (!Utils::Fs::mkpath(m_storageDir))
-                throw AsyncFileStorageError(tr("Could not create directory '%1'.").arg(m_storageDir.toString()));
-
-            auto lockFileDeleter = [](QFile *file)
-            {
-                file->close();
-                file->remove();
-                delete file;
-            };
-            m_lockFile = std::shared_ptr<QFile>(new QFile(lockFilePath.data()), std::move(lockFileDeleter));
-
-            // TODO: This folder locking approach does not work for UNIX systems. Implement it.
-            if (!m_lockFile->open(QFile::WriteOnly))
-                throw AsyncFileStorageError(m_lockFile->errorString());
-
-            lockFile = m_lockFile;
-        }
-        else
-        {
-            m_lockFile = lockFile.lock();
-        }
-    }
+    if (!Utils::Fs::mkpath(m_storageDir))
+        throw AsyncFileStorageError(tr("Could not create directory '%1'.").arg(m_storageDir.toString()));
```

Comment:
Off topic.
Just throwing exceptions seem to be insufficient since the error message won't show up to the user. The error message should also be sent to the logger (with `CRITICAL` log level) so that the user will be notified and able to resolve it.

Reply:
> Just throwing exceptions seem to be insufficient since the error message won't show up to the user.

See #23303.